tests/test_patterns.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2010 Yesudeep Mangalapilly <yesudeep@gmail.com>
+# Copyright 2020 Boris Staletic <boris.staletic@gmail.com>
+
+import pytest
+from pathlib import PureWindowsPath, PurePosixPath
+from watchdog.utils.patterns import _match_path, filter_paths, match_any_paths
+
+
+@pytest.mark.parametrize("input, included_patterns, excluded_patterns, case_sensitive, expected", [
+    ("/users/gorakhargosh/foobar.py", {"*.py"}, {"*.PY"}, True, True),
+    ("/users/gorakhargosh/foobar.py", {"*.py"}, {"*.PY"}, True, True),
+    ("/users/gorakhargosh/", {"*.py"}, {"*.txt"}, False, False),
+    ("/users/gorakhargosh/foobar.py", {"*.py"}, {"*.PY"}, False, ValueError),
```

Comment:
Once I moved the `if case_sensitive` to `_match_path`, I was able to use plain old strings as input as well.